src/plugins/explore/public/application/legacy/discover/application/view_components/canvas/index.tsx

```diff
@@ -186,3 +188,4 @@ export default function DiscoverCanvas({ setHeaderActionMenu, optionalRef }: Vie
 
 const MemoizedDiscoverTable = React.memo(DiscoverTable);
 const MemoizedDiscoverChartContainer = React.memo(DiscoverChartContainer);
+const MemoizedExploreVisualizationContainer = React.memo(ExploreVisualizationContainer);
```

Comment:
lets not memoize this unless we have data: https://kentcdodds.com/blog/usememo-and-usecallback#reactmemo-and-friends

Reply:
i think use memoize here is beneficial because it helps prevent unnecessary re-renders of the VisualizationContainer component when its props (rows and fieldSchema) haven't changed when DiscoverCanvas might re-render frequently due to other state changes so we do not repeat expensive operations such as data transformation and expression generation